crates/rethnet_eth/src/remote.rs

```diff
@@ -97,72 +110,553 @@ impl BlockSpec {
     }
 }
 
-#[derive(serde::Serialize)]
-#[serde(untagged)]
-enum SerializableBlockSpec {
-    /// as a block number
-    #[serde(serialize_with = "serialize_u256")]
-    Number(U256),
-    /// as a block tag (eg "latest")
-    Tag(String),
+#[derive(Clone, Debug, PartialEq)]
+struct ZeroXPrefixedBytes {
+    inner: Bytes,
+}
+
+impl From<Bytes> for ZeroXPrefixedBytes {
+    fn from(b: Bytes) -> Self {
+        ZeroXPrefixedBytes { inner: b }
+    }
 }
 
-impl From<BlockSpec> for SerializableBlockSpec {
-    fn from(block_spec: BlockSpec) -> SerializableBlockSpec {
-        match block_spec {
-            BlockSpec::Number(n) => SerializableBlockSpec::Number(U256::from(n)),
-            BlockSpec::Tag(s) => SerializableBlockSpec::Tag(s),
+impl<'a> serde::Deserialize<'a> for ZeroXPrefixedBytes {
+    fn deserialize<D>(deserializer: D) -> Result<ZeroXPrefixedBytes, D::Error>
+    where
+        D: serde::Deserializer<'a>,
+    {
+        struct ZeroXPrefixedBytesVisitor;
+        impl<'a> serde::de::Visitor<'a> for ZeroXPrefixedBytesVisitor {
+            type Value = ZeroXPrefixedBytes;
+
+            fn expecting(&self, formatter: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+                formatter.write_str("a 0x-prefixed string of hex digits")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                if &value[0..1] == "0x" {
+                    Err(serde::de::Error::custom(
+                        "string does not have a '0x' prefix",
+                    ))
+                } else {
+                    Ok(Bytes::from(
+                        hex::decode(&value[2..])
+                            .unwrap_or_else(|_| panic!("failed to decode hex string \"{value}\"")),
+                    )
+                    .into())
+                }
+            }
         }
+
+        deserializer.deserialize_identifier(ZeroXPrefixedBytesVisitor)
     }
 }
 
-#[derive(serde::Serialize)]
+impl serde::Serialize for ZeroXPrefixedBytes {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&format!("0x{}", hex::encode(self.inner.clone()),))
```

Comment:
Thanks for noticing this. I added the `.clone()` because without it I got error message `cannot move out of `self.inner` which is behind a shared reference`...````move occurs because `self.inner` has type `bytes::Bytes`, which does not implement the `Copy` trait````, and I'm used to reaching for `Clone` when there's no `Copy`, but I didn't realize that a reference would do just fine here. Thanks! Addressed in https://github.com/NomicFoundation/hardhat/pull/3967/commits/0cbf7cc858f83806e7700e0cfb515db05bb5bc5a